### DISCUSSION TITLE
Read me needs a more compelling example

### DISCUSSION THREAD

**Discussion by kevin-j-channon** [carbon-language/carbon-lang#1533] [General]
The "circles" example in the readme doesn't make a great case, since the C++ and Carbon code is nearly identical. Why bother wrangling a whole new language and tool chain for literally no gain? You might as well just write the C++!

I don't know if there is some killer feature currently in Carbon that would allow a great demonstration of "look at this! Just a few lines of Carbon is equivalent to this pile of C++! What are you waiting for!? Get on this train right now."  That sort of thing. Maybe something with strings. Strings are always a ball ache in C++ :)

**Comment by lexi-nadia** [comment]
A couple of things are hidden in that example (like Carbon's use of modules/packages instead of C++ textual headers), but you're right that the two examples don't look much different. But that's the point! :) The fact that "the C++ and Carbon code is nearly identical" is exactly why this example exists. It demonstrates how reading and writing Carbon will still feel a lot like reading and writing C++, just simpler.

**Reply by kevin-j-channon** [reply]
I'd be concerned that those points are a little subtle for this stage in the game. Also, is it simpler to read/write if it's basically the same, but with slightly different names for things and different types of brackets? Surely, it's the same to write?

My point was that if it's basically the same, then you're sort of relying on the "shiny shiny!" factor for people to start using it, which is not really compelling, from a a business point of view.

Maybe additional examples are required, each with specific aims that are very clear to the reader. So, one that shows how some complex C++ can be simplified (my original point) and another that shows how simple interaction with existing C++ code is? There are probably others too. Essentially, at least one example per killer argument that you have for paying the tax of taking up a new language versus just writing the new code in C++.

**Reply by jonmeow** [reply]
The README's example needs to be short due to the context within which it's read; a summary page. Right now, the focus is definitely intended to be on how Carbon is experimental and not ready for use, rather than encouraging developers to use it. Improving the presentation for developers is something I'd expect more focus on when and if the language is ready for developers to write code in Carbon.
